easybuild/easyblocks/f/fdtd_solutions.py

```diff
@@ -33,64 +33,39 @@
+from easybuild.tools.run import run_cmd
 
 
 class EB_FDTD_underscore_Solutions(PackedBinary):
     """Support for building/installing FDTD Solutions."""
 
-    def build_step(self):
-        """No build step for FDTD Solutions."""
-        # locate RPM and rebuild it to make it relocatable
-        rpms = glob.glob(os.path.join(self.cfg['start_dir'], 'rpm_install_files', 'FDTD-%s*.rpm' % self.version))
+    def extract_step(self):
+        """
+        After unpacking the main tar file, we need to unpack the rpm
+        inside it.
+        """
+        super(EB_FDTD_underscore_Solutions, self).extract_step()
+
+        rpms = glob.glob(os.path.join(self.src[0]['finalpath'], 'rpm_install_files', 'FDTD-%s*.rpm' % self.version))
         if len(rpms) != 1:
             raise EasyBuildError("Incorrect number of RPMs found, was expecting exactly one: %s", rpms)
-        rebuilt_dir = os.path.join(self.cfg['start_dir'], 'rebuilt')
-        rebuild_rpm(rpms[0], rebuilt_dir)
+        cmd = "rpm2cpio %s | cpio -idm " % rpms[0]
+        run_cmd(cmd, log_all=True, simple=True)
 
-        # replace original RPM with relocatable RPM
-        rebuilt_rpms = glob.glob(os.path.join(rebuilt_dir, '*', '*.rpm'))
-        if len(rebuilt_rpms) != 1:
-            raise EasyBuildError("Incorrect number of rebuilt RPMs found, was expecting exactly one: %s", rebuilt_rpms)
-
-        try:
-            os.rename(rpms[0], '%s.bk' % rpms[0])
-            shutil.copy2(rebuilt_rpms[0], rpms[0])
-        except OSError, err:
-            raise EasyBuildError("Failed to replace original RPM with rebuilt RPM: %s", err)
+    def build_step(self):
+        """No build step for FDTD Solutions."""
+        pass
 
     def install_step(self):
-        """Install FDTD Solutions using install.sh script."""
-        cmd = "./install.sh"
-        acceptq = ''.join([
-            "If you accept the terms above, type ACCEPT, otherwise press <enter> ",
-            "to cancel the install [REJECT]:"
-        ])
-        qa = {
-            acceptq: "ACCEPT",
-            "Please select an option from the list [1]:": '1',
-            "Please enter the install directory [/opt/lumerical/fdtd]:": self.installdir,
-        }
-        no_qa = [
-        ]
-        std_qa = {
-            "Press <enter> to continue.*": ''
-        }
-        run_cmd_qa(cmd, qa, no_qa=no_qa, std_qa=std_qa, log_all=True, simple=True)
-
-    def make_module_req_guess(self):
-        """Adjust values for PATH, LD_LIBRARY_PATH, etc."""
-        guesses = super(EB_FDTD_underscore_Solutions, self).make_module_req_guess()
-        guesses.update({
-            'PATH': ['opt/lumerical/fdtd/bin'],
-            'LD_LIBRARY_PATH': ['opt/lumerical/fdtd/lib'],
-        })
-        return guesses
+        """Install FDTD Solutions using copy tree."""
+        fdtd_dir = os.path.join(self.cfg['start_dir'], 'opt', 'lumerical', 'fdtd')
+        copy_dir(fdtd_dir, self.installdir, symlinks=self.cfg['keepsymlinks'])
```

Comment:
@akesandgren Installation fails now because the install directory is pre-created, and `copy_dir` doesn't like that.

You'll need to either call `remove_dir(self.installdir)s` first, or override the `make_installdir` method to make it a no-op (just log a message like `"Not pre-creating installation directory %s" % self.installdir` instead)

Reply:
Something like that then?

Reply:
I overlooked that `make_installdir` is also responsible for removing an existing installation directory in case of a forced reinstall, so you should (also) set `self.cfg['dontcreateinstalldir'] = True` before calling the parent:

```python
def make_installdir(self):
    """Override installdir creation"""
    self.log.warning("Not pre-creating installation directory %s" % self.installdir)
    self.cfg['dontcreateinstalldir'] = True
    super(EB_EB_FDTD_underscore_Solutions, self).make_installdir()